tests/Cache/CachingResultTest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Cache;
+
+use Doctrine\Common\Cache\Cache;
+use Doctrine\DBAL\Cache\CachingResult;
+use Doctrine\DBAL\Driver\Result;
+use PHPUnit\Framework\TestCase;
+
+class CachingResultTest extends TestCase
+{
+    /** @var string */
+    private $cacheKey = 'cacheKey';
+
+    /** @var string */
+    private $realKey = 'realKey';
+
+    /** @var int */
+    private $lifetime = 3600;
+
+    /** @var array<string, mixed> */
+    private $resultData = ['id' => 123, 'field' => 'value'];
+
+    /** @var array<string, mixed> */
+    private $cachedData;
+
+    /** @var Cache */
+    private $cache;
+
+    /** @var Result */
+    private $result;
+
+    protected function setUp(): void
+    {
+        $this->result = $this->createMock(Result::class);
+        $this->result->expects(self::exactly(2))
+            ->method('fetchAssociative')
+            ->will($this->onConsecutiveCalls($this->resultData, false));
+
+        $this->cache = $this->createMock(Cache::class);
+        $this->cache->expects(self::exactly(1))
+            ->method('save')
+            ->willReturnCallback(function ($id, $data, $ttl): void {
+                $this->assertEquals($this->cacheKey, $id, 'The cache key should match the given one');
+                $this->assertEquals($this->lifetime, $ttl, 'The cache key ttl should match the given one');
+                $this->cachedData = $data;
+            });
+    }
+
+    public function testShouldSaveResultToCache(): void
+    {
+        $cachingResult = new CachingResult(
+            $this->result,
+            $this->cache,
+            $this->cacheKey,
+            $this->realKey,
+            $this->lifetime,
+            ['otherRealKey' => 'resultValue']
+        );
+
+        do {
+            $row = $cachingResult->fetchAssociative();
+        } while ($row !== false);
```

Comment:
Why does a test need flow control logic? Do we not know how many rows are expected to be fetched here?

Reply:
In this particular test, we know that the 2nd `fetchAssociative` should be `false`. But that is not what is tested here. This block is a standard way to fetch all rows. 

I may use `Doctrine\DBAL\Driver\FetchUtils::fetchAllAssociative` instead of this block if you like.